python/activator/middleware_interface.py

```diff
@@ -689,6 +690,48 @@ def _download(self, remote):
         local.transfer_from(remote, "copy")
         return local
 
+    def check_for_snap(self, instrument: str, groupid: str, snap: int,
+                       detector: int) -> tuple[Snap | None, str | None]:
+        """Check to see if an image for a group/snap is already ingested.
+
+        Parameters
+        ----------
+        instrument : `str`
+        groupid : `str`
+        snap : int
+        detector : int
+
+        Returns
+        -------
+        raw_info : `Snap` or None
+            The expanded information about the image, None if it is not
+            present.
+        oid : `str` or None
+            The pathname of the image, None if it is not present.
+        """
+        datasets = set(self.central_butler.registry.queryDatasets(
+            datasetType="raw",
+            collections=f"{instrument}/raw/all",
+            where="exposure.group_name = g and instrument = i"
+                  " and exposure.seq_start - exposure.seq_num = s"
+                  " and detector = d",
+            bind=dict(g=groupid, i=instrument, s=snap, d=detector)
+        ))
+        if not datasets:
+            return None, None
+        return (
+            Snap(
+                instrument=instrument,
+                detector=detector,
+                group=groupid,
+                snap=snap,
+                exp_id=datasets[0].dataId["exposure"],
+                filter=datasets[0].dataId["physical_filter"],
+            ),
+            self.central_butler.getURI(datasets[0], collections=datasets[0].run)
+                .path.removeprefix("/")
```

Comment:
No longer there.  `removeprefix` was because oids work best as relative paths, not absolute (that way they can be attached to URLs like "s3://bucket-name/").